contributed_definitions/NXregion.nxdl.xml

```diff
@@ -0,0 +1,193 @@
+						@signal = "sum"
+						sum[60]
+
+		the ``sum`` dataset contains the summed areas in each frame.
+		Another example, a hyperspectral image downsampled 16-fold in energy::
+
+			detector:NXdetector/
+				data[128,128,4096]
+				region:NXregion/
+					@type  = "rectangular"
+					parent = "data"
+					start  = [2]
+					count  = [20]
+					stride = [32]
+					block  = [16]
+					downsampled:NXdata/
+						@signal = "maximum"
+						@auxiliary_signals = "copy"
+						maximum[128,128,20]
+						copy[128,128,320]
+
+		the ``copy`` dataset selects 20 16-channel blocks that start 32 channels apart,
+		the ``maximum`` dataset will show maximum values in each 16-channel block
+		in every spectra.
+	</doc>
+	<attribute name="type" optional="false">
+		<doc>
+			This is ``rectangular`` to describe the region as a hyper-rectangle in
+			the index space of its parent group's data field
+		</doc>
+		<enumeration>
+			<item value="rectangular" />
+		</enumeration>
+	</attribute>
+	<field name="parent">
+		<doc>
+			The name of data field in the parent group or the path of a data field relative
+			to the parent group (so it could be a field in subgroup)
+		</doc>
+	</field>
+	<field name="parent_mask">
+		<doc>
+			The name of an optional mask field in the parent group whose shape matches the
+			region dimensions of the parent data field, i.e. :math:`\boldsymbol{d}`. For example,
+			this could be ``pixel_mask`` of an :ref:`NXdetector`
+		</doc>
+	</field>
+	<field name="start" type="NX_NUMBER" recommended="true">
+		<doc>
+			The starting position for region in detector data field array.
+			This is recommended as it also defines the region rank.
+			If omitted then defined as an array of zeros
+		</doc>
+		<dimensions rank="1">
+			<dim index="1" value="R"/>
+		</dimensions>
+	</field>
+	<field name="count" type="NX_INT" recommended="true">
+		<doc>
+			The number of blocks or items in hyperslab selection.
+			If omitted then defined as an array of dimensions that take into account
+			the other hyperslab selection fields to span the parent data field's shape
+		</doc>
+		<dimensions rank="1">
+			<dim index="1" value="R"/>
+		</dimensions>
+	</field>
+	<field name="stride" type="NX_INT">
+		<doc>
+			An optional field to define striding used to downsample data.
+			If omitted then defined as an array of ones
+		</doc>
+		<dimensions rank="1">
+			<dim index="1" value="R"/>
+		</dimensions>
+	</field>
+	<field name="block" type="NX_INT">
+		<doc>
+			An optional field to define the block size used to copy or downsample data. In the
+			:math:`i`-th dimension, if :math:`\mathbf{block}[i] \lt \mathbf{stride}[i]`
+			then the downsampling blocks have gaps between them; when ``block`` matches ``stride``
+			then the blocks are contiguous; otherwise the blocks overlap.
+			If omitted then defined as an array of ones
+		</doc>
+		<dimensions rank="1">
+			<dim index="1" value="R"/>
+		</dimensions>
+	</field>
+	<field name="scale" type="NX_NUMBER">
```

Comment:
Ok so for example suppose we have an 8-bit image, we do 2x2 summing and we want the result to still be 8 bit. In that case we need `scale=4` to not have any value overflow. Is that the use case? Then why not doing 2x2 averaging instead of 2x2 summing?

Anyway, I think I understand now. Thanks!

Reply:
Because it could be another number and not match the size of each block. For example, see areaDetector's [NDPluginROIStat](https://areadetector.github.io/master/ADCore/NDPluginROIStat.html).